api/controllers/budgetControllers.js

```diff
@@ -0,0 +1,35 @@
+const Budget = require('../models/budget');
+const Expense = require('../models/expense');
+
+const BudgetController = {
+    create: async (req, res) => {
+        try {
+            const { title, budgetAmount } = req.body;
+            const budget = await new Budget({ title, budgetAmount }).save();
+            res.json(budget);
+        } catch (error) {
+            console.log('BudgetController@create: ', error);
+            res.status(422).json({ error });
+        }
+    },
+    getSummary: async (req, res) => {
+        try {
+            const { id } = req.params;
+            const budget = await Budget.findById(id).exec();
+            const expenses = await Expense.aggregate([
+                { $match: { budget: budget._id } },
+                { $group: { _id: budget._id,  total: { $sum: '$amount' } }
+                }
+            ]).exec();
+            
+            const { total } = expenses[0];
+            const { budgetAmount } = budget;
+            res.json({ budgetLeft: budgetAmount - total})
+        } catch (error) {
+            console.log('BudgetController@getSummary', error);
```

Comment:
I would think that you need to provide a place for Node to stop here. Console logging may not be sufficient enough to terminate the process, thus leaving your node server hanging.. awaiting a `response/status code`